_rules/aria-required-id-references-in6db8.md

```diff
@@ -0,0 +1,199 @@
+<div
+	role="scrollbar"
+	aria-controls="content"
+	aria-orientation="vertical"
+	aria-valuemax="100"
+	aria-valuemin="0"
+	aria-valuenow="25"
+></div>
+```
+
+#### Passed Example 2
+
+The `aria-controls` [attribute value][] of this expanded `combobox` matches the `id` of the `main` element in the same document.
+
+```html
+<label for="tag_combo">Tag</label>
+<input
+	type="text"
+	id="tag_combo"
+	role="combobox"
+	aria-expanded="true"
+	aria-controls="popup_listbox"
+	aria-activedescendant="selected_option"
+/>
+<ul role="listbox" id="popup_listbox">
+	<li role="option">Zebra</li>
+	<li role="option" id="selected_option">Zoom</li>
+</ul>
+```
+
+#### Passed Example 3
+
+The `aria-controls` [attribute value][] of this `scrollbar` has two IDs. The `content-2` ID matches the `id` of the `main` element in the same document.
+
+```html
+<main id="content-2">Lorem ipsum...</main>
+<div
+	role="scrollbar"
+	aria-controls="content-1 content-2"
+	aria-orientation="vertical"
+	aria-valuemax="100"
+	aria-valuemin="0"
+	aria-valuenow="25"
+></div>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The `aria-controls` attribute of this expanded `combobox` references an ID of `popup_listbox` which does not exist in the document.
+
+```html
+<label>
+	Tag
+	<input role="combobox" aria-expanded="true" aria-controls="popup_listbox" />
+</label>
+```
+
+#### Failed Example 2
+
+The `aria-controls` attribute of this `scrollbar` references IDs of `content-1` and `content-2`. Neither of these IDs exist in the document.
+
+```html
+<main>Lorem ipsum...</main>
+<div
+	role="scrollbar"
+	aria-controls="content-1 content-2"
+	aria-orientation="vertical"
+	aria-valuemax="100"
+	aria-valuemin="0"
+	aria-valuenow="25"
+></div>
+```
+
+#### Failed Example 3
+
+The `aria-controls` attribute of this expanded `combobox` references a `popup_listbox` ID. This `id` exists, but in a different DOM tree as the `combobox`.
+
+```html
+<aria-listbox>
+	<label for="tag_combo">Tag</label>
+	<input
+		type="text"
+		id="tag_combo"
+		role="combobox"
+		aria-expanded="true"
+		aria-controls="popup_listbox"
+		aria-activedescendant="selected_option"
+	/>
+
+	<template shadowrootmode="open">
+		<slot></slot>
+		<ul role="listbox" id="popup_listbox">
+			<li role="option">Zebra</li>
+			<li role="option" id="selected_option">Zoom</li>
+		</ul>
+	</template>
+</aria-listbox>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+The `combobox` does not have an `aria-expanded` [attribute value][] of `true`.
+
+```html
+<label for="tag_combo">Tag</label>
+<input type="text" id="tag_combo" role="combobox" aria-expanded="false" aria-controls="popup_listbox" />
+```
+
+#### Inapplicable Example 2
+
+This element with an `aria-controls` attribute is not a [semantic][] `scrollbar` nor `combobox`.
+
+```html
+<button aria-controls="my-modal">Open the modal</button>
+```
+
+#### Inapplicable Example 3
+
+The `button` does not have an `aria-controls` attribute.
```

Comment:
```suggestion
There is no `aria-controls` attribute.
```
I also agree with @tombrunet that this should maybe be the first Inapplicable example since it seems simpler than the rest.